cpp/src/io/parquet/reader_impl_chunking.cu

```diff
@@ -253,13 +253,15 @@ struct set_row_index {
   device_span<ColumnChunkDesc const> chunks;
   device_span<PageInfo const> pages;
   device_span<cumulative_page_info> c_info;
+  size_t const max_row;
```

Comment:
I'm sure @vuule can help us with the link, but const inside a struct is not advised. It means we can't use move semantics on the struct. I don't know if `set_row_index` is passed around though.

Reply:
Typically I would argue to always use `const` for member variables, because the compiler can apply load optimization on it. For example, if the variable is constant, the compiler can load it into shared memory for all threads thus improving the kernel performance. Otherwise, every thread may always need to load it whenever they need to use it. On the other hand, copying the struct for just a couple of times is cheap.

Reply:
I've stopped bringing this one up in recent PRs, folks are resistant and the benefit is very minor in the first place.
FWIW it's fine in a functor, we don't rely on a cheap move here.

Reply:
> For example, if the variable is constant, the compiler can load it into shared memory for all threads thus improving the kernel performance.

Do we know if this really happens? C++ has `const_cast`, so `const` does not really guarantee immutability.